plugin-server/src/main/ingestion-queues/batch-processing/buffer.ts

```diff
@@ -0,0 +1,61 @@
+import { EachBatchPayload, KafkaMessage } from 'kafkajs'
```

Comment:
File name and main function don't line up.

Suggestion: Let's use the naming pattern `each-batch-buffer.ts`?